src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1306,14 +1308,10 @@ private int getMostEfficientTool(@NotNull final Block target)
         {
             final ItemStack item = inventory.getStackInSlot(i);
             final int level = Utils.getMiningLevel(item, tool);
-
-            if (level >= required && level < bestLevel)
+            if (tool == null || InventoryUtils.verifyToolLevel(item, level, required, hutLevel))
             {
-                if (tool == null || InventoryUtils.verifyToolLevel(item, level, hutLevel))
-                {
-                    bestSlot = i;
-                    bestLevel = level;
-                }
+                bestSlot = i;
+                bestLevel = level;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "bestLevel". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1854)